### ISSUE TITLE
Step 2 of price list upload fails if user leaves date field blank

### ISSUE THREAD

**Issue by toolness** [18F/calc#705]
I think we need to make the fields required or something?

<details>
  <summary>Full stack trace</summary>
```
Environment:


Request Method: POST
Request URL: http://localhost:8000/data-capture/step/2

Django Version: 1.8.14
Python Version: 3.4.5
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.humanize',
 'whitenoise.runserver_nostatic',
 'django.contrib.staticfiles',
 'debug_toolbar',
 'django_rq',
 'data_explorer',
 'contracts',
 'data_capture.apps.DefaultDataCaptureApp',
 'api',
 'djorm_pgfulltext',
 'rest_framework',
 'corsheaders',
 'uaa_client',
 'styleguide',
 'meta',
 'frontend',
 'fake_uaa_provider')
Installed Middleware:
('hourglass.middleware.ComplianceMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'hourglass.middleware.DebugOnlyDebugToolbarMiddleware')


Traceback:
File "/usr/local/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  132.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/calc/data_capture/decorators.py" in wrapper
  46.                 return function(request, *args, **kwargs)
File "/usr/local/lib/python3.4/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  22.                 return view_func(request, *args, **kwargs)
File "/calc/data_capture/views/price_list_upload.py" in step_2
  73.                 'contract_start': ymd(form.cleaned_data['contract_start']),
File "/calc/data_capture/views/price_list_upload.py" in ymd
  57.         return date_input.strftime("%Y-%m-%d")

Exception Type: AttributeError at /data-capture/step/2
Exception Value: 'NoneType' object has no attribute 'strftime'
```
</details>

**Comment by toolness** [comment]
I'm doing some refactoring on the code around here so I'll hopefully be able to take care of this too. Assigning to myself!

**Comment by jseppi** [comment]
yup, it should be required per #498

**Comment by toolness** [comment]
I've fixed this in #713, but not by fixing #498--rather, it just restores our pre-#583 behavior of having the dates be optional.  We can resolve #498 in a separate PR since #713 is already a bit hefty as-is!
